File: app/src/main/java/com/oryanmat/trellowidget/widget/WidgetService.java
package com.oryanmat.trellowidget.widget;

import android.appwidget.AppWidgetManager;
import android.content.Intent;
import android.widget.RemoteViewsService;

public class WidgetService extends RemoteViewsService {
    @Override
    public RemoteViewsFactory onGetViewFactory(Intent intent) {
        int widgetId = intent.getIntExtra(
                AppWidgetManager.EXTRA_APPWIDGET_ID,
                AppWidgetManager.INVALID_APPWIDGET_ID);
        return new CardRemoteViewFactory(this, widgetId);
    }
}

File: app/src/main/kotlin/com/github/oryanmat/trellowidget/widget/WidgetService.kt
package com.github.oryanmat.trellowidget.widget

import android.appwidget.AppWidgetManager
import android.content.Intent
import android.widget.RemoteViewsService

class WidgetService : RemoteViewsService() {
    override fun onGetViewFactory(intent: Intent) =
            CardRemoteViewFactory(this, intent.getIntExtra(
                    AppWidgetManager.EXTRA_APPWIDGET_ID,
                    AppWidgetManager.INVALID_APPWIDGET_ID))
}